modules/viz/include/opencv2/viz/widgets.hpp

```diff
@@ -842,7 +930,833 @@ namespace cv
+*/
+        struct CV_EXPORTS_W_SIMPLE PyWCameraPosition
+        {
+        public:
+            /** @brief Creates camera coordinate frame at the origin.
+
+            ![Camera coordinate frame](images/cpw1.png)
+             */
+            CV_WRAP PyWCameraPosition(double scale = 1.0)
+            {
+                widget = cv::makePtr<cv::viz::WCameraPosition>(scale);
+            }
+            /** @brief Display the viewing frustum
+            @param K Intrinsic matrix of the camera or fov Field of view of the camera (horizontal, vertical).
+            @param scale Scale of the frustum.
+            @param color Color of the frustum.
+
+            Creates viewing frustum of the camera based on its intrinsic matrix K.
+
+            ![Camera viewing frustum](images/cpw2.png)
+            */
+            CV_WRAP PyWCameraPosition(InputArray  K, double scale = 1.0, const PyColor &color = PyColor(255, 255, 255))
+            {
+                if (K.kind() == _InputArray::MAT)
+                {
+                    Mat k = K.getMat();
+                    if (k.rows == 3 && k.cols == 3)
+                    {
+                        Matx33d x = k;
+                        widget = cv::makePtr<cv::viz::WCameraPosition>(x, scale, color.c);
+
+                    }
+                    else if (k.rows==2 && k.cols==1)
+                        widget = cv::makePtr<cv::viz::WCameraPosition>(Vec2d(k.at<double>(0,0), k.at<double>(0,1)), scale, color.c);
+                    else
+                        CV_Error(-5, "unknown size");
+                }
+                else
+                    CV_Error(-5, "unknown type");
+            }
+
+            /** @brief Display image on the far plane of the viewing frustum
+
+            @param K Intrinsic matrix of the camera.
+            @param image BGR or Gray-Scale image that is going to be displayed on the far plane of the frustum.
+            @param scale Scale of the frustum and image.
+            @param color Color of the frustum.
+
+            Creates viewing frustum of the camera based on its intrinsic matrix K, and displays image on
+            the far end plane.
+
+            ![Camera viewing frustum with image](images/cpw3.png)
+             */
+            CV_WRAP PyWCameraPosition(InputArray K, InputArray image, double scale = 1.0, const PyColor &color = PyColor(255, 255, 255))
+            {
+                if (K.kind() == _InputArray::MAT)
+                {
+                    Mat k = K.getMat();
+                    if (k.rows == 3 && k.cols == 3)
+                    {
+                        Matx33d x = k;
+                        widget = cv::makePtr<cv::viz::WCameraPosition>(x, image, scale, color.c);
+
+                    }
+                    else if (k.rows == 2 && k.cols == 1)
+                        widget = cv::makePtr<cv::viz::WCameraPosition>(Vec2d(k.at<double>(0, 0), k.at<double>(0, 1)), image, scale, color.c);
```

Comment:
There are several exceptions from Python tests here (in debug builds).

It is better to replace:
- check: `else if (k.total() == 2)`
- use `.at()` with single arg: `k.at<double>(0), k.at<double>(1)`

Causes by "undefined" 1D layout represented as 2D Mat (it may be row-based or column-based).

---

Please replace similar cases of `.at()` usage of 1D arrays below (use single argument form).